XmlDocConflictResolver/ConflictChecker.cs

```diff
@@ -0,0 +1,323 @@
+                //    }
+                //    catch (Exception ex)
+                //    {
+                //        Log.Error($"Failed to load '{fileInfo.FullName}'. {ex}");
+                //    }
+
+                //    if (xDoc != null && encoding != null)
+                //    {
+                //        DocsComments.LoadDocsFile(xDoc, fileInfo.FullName, encoding);
+                //    }
+                //}
+                //Log.Success("Finished looking for Docs xml files.");
+                //Log.Line();
+            }
+        }
+
+        internal void Start()
+        {
+            if (!IntelliSenseXmlComments.Members.Any())
+            {
+                Log.Error("No IntelliSense XML comments found.");
+                return;
+            }
+
+            if (!DocsComments.Types.Any())
+            {
+                Log.Error("No docs type APIs found.");
+                return;
+            }
+
+            InsertConflictingText();
+        }
+
+        private void InsertConflictingText()
+        {
+            Log.Info("Looking for IntelliSense xml comments that differ from the docs...");
+
+            foreach (IntelliSenseXmlMember member in IntelliSenseXmlComments.Members.Values)
+            {
+                CheckForConflictingTextForMember(member);
+            }
+        }
+
+        private void CheckForConflictingTextForMember(IntelliSenseXmlMember ixmlMember)
+        {
+            bool foundDocsApi;
+            IDocsAPI? ecmaxmlApi;
+
+            // Find docs type or member.
+            if (ixmlMember.IsType())
+                foundDocsApi = DocsComments.Types.TryGetValue(ixmlMember.Name, out ecmaxmlApi);
+            else
+                foundDocsApi = DocsComments.Members.TryGetValue(ixmlMember.Name, out ecmaxmlApi);
+
+            if (foundDocsApi && ecmaxmlApi != null)
+            {
+                if (!ecmaxmlApi.Summary.IsDocsEmpty() &&
+                    String.Compare(ecmaxmlApi.Summary, ixmlMember.Summary) != 0)
+                {
+                    ixmlMember.Summary = ecmaxmlApi.Summary;
+                    ixmlMember.XmlFile.Changed = true;
+                }
+
+                if (!ecmaxmlApi.Returns.IsDocsEmpty() &&
+                    String.Compare(ecmaxmlApi.Returns, ixmlMember.Returns) != 0)
+                {
+                    ixmlMember.Returns = ecmaxmlApi.Returns;
+                    ixmlMember.XmlFile.Changed = true;
+                }
+
+                //foreach (IntelliSenseXmlParam ixmlParam in ixmlMember.Params)
+                for (int i = 0; i < ixmlMember.Params.Count; i++)
+                {
+                    IntelliSenseXmlParam ixmlParam = ixmlMember.Params[i];
+
+                    // Find matching param from docs XML.
+                    if (!ixmlParam.Value.IsDocsEmpty())
+                    {
+                        DocsParam? ecmaxmlParam = ecmaxmlApi.Params.Where(p => p.Name == ixmlParam.Name).FirstOrDefault();
```

Comment:
```suggestion
                        DocsParam? ecmaxmlParam = ecmaxmlApi.Params.FirstOrDefault(p => p.Name == ixmlParam.Name);
```

An empty `FirstOrDefault` preceded by a `Where` clause is a code smell to me. The first or default method takes a filter predicate, simplify this by removing the where clause.